lightningd/channel_control.c

```diff
@@ -1631,6 +1631,9 @@ bool peer_start_channeld(struct channel *channel,
 	initmsg = towire_channeld_init(tmpctx,
 				       chainparams,
 				       ld->our_features,
+				       /* Capabilities arg needs to be a tal array */
+				       tal_dup_arr(tmpctx, u32, ld->hsm_capabilities,
```

Comment:
Weird... since you use `tal_count` then ld->hsm_capabilities must be a tal array, so you shouldn't need to copy it?